core/src/main/java/io/cucumber/core/api/TypeRegistryConfigurer.java

```diff
@@ -6,8 +6,12 @@
 
 /**
  * The type registry configurer allows to configure a new type registry and the locale.
+ * @deprecated Please use annotation based configuration.
+ * See <a href="https://github.com/cucumber/cucumber-jvm/blob/master/examples/java-calculator/src/test/java/io/cucumber/examples/java/ShoppingSteps.java">Annotation based example</a>
+ * See <a href="https://github.com/cucumber/cucumber-jvm/blob/master/examples/java8-calculator/src/test/java/io/cucumber/examples/java8/ShoppingSteps.java">Lambda based example</a>
```

Comment:
Ideally this should link to the documentation pages rather then github examples.

Reply:
I can't really change configuration page in the docs repo before v5 release. This could mislead users.

So I'll change this links to https://cucumber.io/docs/cucumber/configuration/ after v5 release.

Reply:
We also don't have to introduce this in 5.0.0. I'd rather have the docs in order so there is a clear way out before we deprecate it.

Reply:
So. I was planning to do this in in v5.x rather then v5.0 because it would make updating the documentation a bit easier but since we're working with release candidates now I think we can prepare the MR for v5.0 along with the release of v5.0.